File: MusicShop.Model/Synthesizer.cs
using MusicShop.Model.BaseModel;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicShop.Model
{
    public class Synthesizer : BaseProduct
    {
        public int? KeyboardSize { get; set; }
        public bool? WeighedKeys { get; set; }
        public int? Polyphony { get; set; }
        public int? NumberOfPresets { get; set; }
    }
}

File: UI/musicshop_mobile/lib/models/synthesizer/synthesizer.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:musicshop_mobile/models/Abstract/product.dart';
import 'package:musicshop_mobile/models/brand/brand.dart';

part 'synthesizer.g.dart';

@JsonSerializable()
class Synthesizer {
  int? id;
  String? productNumber;
  String? productImage;
  Brand? brand;
  String? model;
  double? price;
  String? description;
  DateTime? createdAt;
  DateTime? updatedAt;
  int? keyboardSize;
  bool? weighedKeys;
  int? polyphony;
  int? numberOfPresets;

  Synthesizer();

  factory Synthesizer.fromJson(Map<String, dynamic> json) =>
      _$SynthesizerFromJson(json);

  Map<String, dynamic> toJson() => _$SynthesizerToJson(this);
}
